### ISSUE TITLE
Error when using exported detection yolov10 model for text detection PaddleOCR

### ISSUE THREAD

**Issue by jtong99** [PaddlePaddle/PaddleOCR#14346]
### 🔎 Search before asking

- [X] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.
- [X] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.
- [X] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### 🐛 Bug (问题描述)

Hi, I have a trained model for text detection from yolov10, I used export model function from yolo and export to Paddle format and I got 3 files:

```
inference.pdiparams
inference.pdiparams.info
inference.pdmodel
```

After that I tested detection inference using exported model with this command:

```
python3 tools/infer/predict_det.py --det_model_dir="./inference/yolov10_model/" --image="/home/jovyan/PaddleOCR/images/image.png" --use_gpu=True 
```

I got this error:

```
[2024/12/07 11:18:46] ppocr WARNING: The first GPU is used for inference by default, GPU ID: 0
Traceback (most recent call last):
  File "/home/jovyan/PaddleOCR/tools/infer/predict_det.py", line 440, in <module>
    dt_boxes, _ = text_detector(img)
                  ^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/PaddleOCR/tools/infer/predict_det.py", line 396, in __call__
    dt_boxes, elapse = self.predict(img)
                       ^^^^^^^^^^^^^^^^^
  File "/home/jovyan/PaddleOCR/tools/infer/predict_det.py", line 255, in predict
    self.predictor.run()
ValueError: (InvalidArgument) The 'shape' in ReshapeOp is invalid. The input tensor X'size must be equal to the capacity of 'shape'. But received X's shape = [1, 512, 30, 21], X's size = 322560, 'shape' is [1, 4, 128, 400], the capacity of 'shape' is 204800.
  [Hint: Expected capacity == in_size, but received capacity:204800 != in_size:322560.] (at /paddle/paddle/phi/infermeta/unary.cc:1967)
  [operator < reshape2 > error]
```

Can anybody face with this problem and have solution?

Thanks

### 🏃‍♂️ Environment (运行环境)

```
Ubuntu 20
Environment: Jupyter Python 3.8.10
```

### 🌰 Minimal Reproducible Example (最小可复现问题的Demo)

```
python3 tools/infer/predict_det.py --det_model_dir="./inference/yolov10_model/" --image="/home/jovyan/PaddleOCR/images/image.png" --use_gpu=True 
```

**Comment by GreatV** [comment]
Currently, PaddleOCR’s text detection modules are designed and optimized specifically for text detection tasks, relying on models such as DB, EAST, and SAST. These models differ significantly from YOLO in terms of network architecture, output formats, and post-processing logic. As a result, PaddleOCR’s existing inference scripts (e.g., `predict_det.py`) are not directly compatible with YOLO model outputs.

If you wish to use a YOLO-based model for text detection within the PaddleOCR pipeline, you will need to:

1. **Custom Inference Script:** Write a custom inference script to load and run your YOLO model. This script should also handle the conversion of YOLO’s output—typically bounding boxes and confidence scores—into a format that aligns more closely with text detection tasks.

2. **Post-Processing Adaptation:** Implement additional post-processing steps to transform the YOLO detection results into a format suitable for text detection. Text detection often requires different preprocessing compared to generic object detection, such as filtering out non-text regions or conforming to specific text line or polygonal output formats.

3. **Integration with OCR Pipeline:** Once you have text-compatible detection results, you can feed these results into the PaddleOCR recognition models. This enables a fully integrated OCR workflow, albeit with some custom code.

In short, PaddleOCR does not natively support YOLO model structures at this time. You will need to adapt and integrate YOLO-based text detection models through custom scripts and processing steps.

**Comment by jtong99** [comment]
> Currently, PaddleOCR’s text detection modules are designed and optimized specifically for text detection tasks, relying on models such as DB, EAST, and SAST. These models differ significantly from YOLO in terms of network architecture, output formats, and post-processing logic. As a result, PaddleOCR’s existing inference scripts (e.g., `predict_det.py`) are not directly compatible with YOLO model outputs.
> 
> If you wish to use a YOLO-based model for text detection within the PaddleOCR pipeline, you will need to:
> 
> 1. **Custom Inference Script:** Write a custom inference script to load and run your YOLO model. This script should also handle the conversion of YOLO’s output—typically bounding boxes and confidence scores—into a format that aligns more closely with text detection tasks.
> 2. **Post-Processing Adaptation:** Implement additional post-processing steps to transform the YOLO detection results into a format suitable for text detection. Text detection often requires different preprocessing compared to generic object detection, such as filtering out non-text regions or conforming to specific text line or polygonal output formats.
> 3. **Integration with OCR Pipeline:** Once you have text-compatible detection results, you can feed these results into the PaddleOCR recognition models. This enables a fully integrated OCR workflow, albeit with some custom code.
> 
> In short, PaddleOCR does not natively support YOLO model structures at this time. You will need to adapt and integrate YOLO-based text detection models through custom scripts and processing steps.

Thanks for your answer. Seems like it takes a lot of things to modify the current inference if I use converted model from Yolo. It will be easier if I make a pipeline for this. Thanks again.
